syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.openk9.tenantmanager.grpc";
option java_outer_classname = "TenantManagerProto";
import "google/protobuf/empty.proto";

package grpc;

service TenantManager {
  rpc FindTenant(TenantRequest) returns (TenantResponse) {}
  rpc FindTenantList(google.protobuf.Empty) returns (TenantListResponse) {}
}

message TenantRequest {
  string virtualHost = 1;
}

message TenantListResponse {
  repeated TenantResponse tenantResponse = 1;
}

message TenantResponse {
  string schemaName = 1;
  string virtualHost = 2;
  string clientId = 3;
  optional string clientSecret = 4;
  string realmName = 5;
}